src/main/java/org/jenkins/plugins/lockableresources/GetLockStep.java

```diff
@@ -0,0 +1,169 @@
+package org.jenkins.plugins.lockableresources;
+
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.AutoCompletionCandidates;
+import hudson.model.TaskListener;
+import hudson.util.FormValidation;
+import org.jenkins.plugins.lockableresources.queue.LockableResourcesStruct;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class GetLockStep extends Step {
+    private static final Logger LOGGER = Logger.getLogger(GetLockStep.class.getName());
+
+    @CheckForNull
+    private String resource;
+
+    @CheckForNull
+    private String label;
+
+    private int quantity = 0;
+
+    @DataBoundConstructor
+    public GetLockStep() {
+    }
+
+    @DataBoundSetter
+    public void setResource(String resource) {
+        this.resource = Util.fixEmpty(resource);
+    }
+
+    public String getResource() {
+        return resource;
+    }
+
+    @DataBoundSetter
+    public void setLabel(String label) {
+        this.label = Util.fixEmpty(label);
+    }
+
+    public String getLabel() {
+        return label;
+    }
+
+    @DataBoundSetter
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    @Override
+    public StepExecution start(StepContext context) throws Exception {
+        return new Execution(this, context);
+    }
+
+    /**
+     * Label and resource are mutual exclusive.
+     */
+    public void validate() throws Exception {
+        if (label != null && !label.isEmpty() && resource !=  null && !resource.isEmpty()) {
+            throw new IllegalArgumentException("Label and resource name cannot be specified simultaneously.");
+        }
+    }
+
+
+
+    @Extension
+    public static class DescriptorImpl extends StepDescriptor {
+        @Override
+        public String getFunctionName() {
+            return "getLock";
```

Comment:
This step will be inevitably available as a primitive step (also listed in the snippet generator). I think this is going to cause confusion (in addition to the confusion already generated by the "concurrency suite" (milestone + lock)).

Agree with @jglick, the only primitive should be the block scoped `lock`. Couldn't be this handle in declarative itself without defining a new primitive?